File: firebase/functions/src/interfaces/ticket.ts
export interface Ticket {
  _type: 'ticket';
  id: number;
  slug: string;
  unique_url: string;
  company_name?: string;
  email: string;
  first_name: string;
  last_name: string;
  name: string;
  reference: string;
  state: 'complete' | 'incomplete';
  test_mode: boolean;
  registration_id: number;
  release_id: number;
  avatar_url: string;
  void: boolean;
  changes_locked: boolean;
  consented_at?: string | null;
  discount_code_used?: string | null;
  created_at: string;
  updated_at: string;
  assigned: boolean;
  qr_url: string;
  show_qr_code: boolean;
  qr_code_disabled: boolean;
}

File: example/lib/core/src/models/ticket.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'ticket.freezed.dart';
part 'ticket.g.dart';

@freezed
class Ticket with _$Ticket {
  const factory Ticket({
    required int id,
    required String? slug,
    required String reference,
    @JsonKey(name: 'qr_url') required String qr,
    required String name,
    required String email,
    @JsonKey(name: 'company_name') String? companyName,
    @JsonKey(name: 'unique_url') String? uniqueUrl,
    @JsonKey(name: 'avatar_url') required String avatarUrl,
    required bool? redeemed,
  }) = _Ticket;

  factory Ticket.fromJson(Map<String, Object?> json) => _$TicketFromJson(json);
}
